@here/harp-mapview/lib/geometry/TileGeometryCreator.ts

```diff
@@ -1368,48 +1367,29 @@ export class TileGeometryCreator {
         return plane;
     }
 
-    /**
-     * Pass the feature data on to the object, so it can be used in picking
-     * `MapView.intersectMapObjects()`. Do not pass the feature data if the technique is a
-     * solid-line, because the line picking functionality for the lines is not object based, but
-     * tile based.
-     *
-     * @param srcGeometry The original [[Geometry]].
-     * @param technique The corresponding [[Technique]].
-     * @param object The object to pass info to.
-     */
-    private addFeatureData(srcGeometry: Geometry, technique: Technique, object: THREE.Object3D) {
-        if (
-            ((srcGeometry.objInfos !== undefined && srcGeometry.objInfos.length > 0) ||
-                isCirclesTechnique(technique) ||
-                isSquaresTechnique(technique)) &&
-            !isSolidLineTechnique(technique)
-        ) {
-            const featureData: TileFeatureData = {
-                geometryType: srcGeometry.type,
-                starts: srcGeometry.featureStarts,
-                objInfos: srcGeometry.objInfos
-            };
-            object.userData.feature = featureData;
-        }
-    }
-
-    private addGeometryObjInfos(
+    private addUserData(
         tile: Tile,
         srcGeometry: Geometry,
         technique: Technique,
         object: THREE.Object3D
     ) {
-        if (srcGeometry.objInfos === undefined || Object.keys(object.userData).length > 0) {
-            return;
-        }
+        const hasObjInfos = (srcGeometry.objInfos?.length ?? 0) > 0;
 
         if (isTerrainTechnique(technique)) {
+            assert(
+                Object.keys(object.userData).length === 0,
+                "Unexpected user data in terrain object"
+            );
+
             if (typeof srcGeometry.objInfos![0] === "number") {
                 assert(false, "Wrong attribute map type for terrain geometry");
                 return;
             }
 
+            if (!hasObjInfos) {
```

Comment:
I see that entire block has no sense to perform if hasObjInfos === false, otherwise it simply catches some assertions (but those are not extracted from JS code, so debugging will be always on).
Summarizing it would be the best just to move this:
```
if (!hasObjInfos) {
    return;
}
```
at the very beginning of the block.

Reply:
done